lib/std/ordsets.nim

```diff
@@ -0,0 +1,782 @@
+    assert difference(a, b).len == 2
+    ## {1, 2}
+
+  result = initOrdSet[A]()
+  for item in s1:
+    if not contains(s2, item):
+      incl(result, item)
+
+proc symmetricDifference*[A](s1, s2: OrdSet[A]): OrdSet[A] =
+  ## Returns the symmetric difference of the sets `s1` and `s2`.
+  runnableExamples:
+    var
+      a = initOrdSet[int]()
+      b = initOrdSet[int]()
+    a.incl(1); a.incl(2); a.incl(3)
+    b.incl(3); b.incl(4); b.incl(5)
+    assert symmetricDifference(a, b).len == 4
+    ## {1, 2, 4, 5}
+
+  result.assign(s1)
+  for item in s2:
+    if containsOrIncl(result, item): excl(result, item)
+
+proc `+`*[A](s1, s2: OrdSet[A]): OrdSet[A] {.inline.} =
+  ## Alias for `union(s1, s2) <#union,OrdSet[A],OrdSet[A]>`_.
+  result = union(s1, s2)
+
+proc `*`*[A](s1, s2: OrdSet[A]): OrdSet[A] {.inline.} =
+  ## Alias for `intersection(s1, s2) <#intersection,OrdSet[A],OrdSet[A]>`_.
+  result = intersection(s1, s2)
+
+proc `-`*[A](s1, s2: OrdSet[A]): OrdSet[A] {.inline.} =
+  ## Alias for `difference(s1, s2) <#difference,OrdSet[A],OrdSet[A]>`_.
+  result = difference(s1, s2)
+
+proc disjoint*[A](s1, s2: OrdSet[A]): bool =
+  ## Returns true if the sets `s1` and `s2` have no items in common.
+  runnableExamples:
+    var
+      a = initOrdSet[int]()
+      b = initOrdSet[int]()
+    a.incl(1); a.incl(2)
+    b.incl(2); b.incl(3)
+    assert disjoint(a, b) == false
+    b.excl(2)
+    assert disjoint(a, b) == true
+
+  for item in s1:
+    if contains(s2, item):
+      return false
+  return true
+
+proc card*[A](s: OrdSet[A]): int {.inline.} =
+  ## Alias for `len() <#len,OrdSet[A]>`_.
+  result = s.len()
+
+proc `<=`*[A](s1, s2: OrdSet[A]): bool =
+  ## Returns true if `s1` is subset of `s2`.
+  ##
+  ## A subset `s1` has all of its elements in `s2`, and `s2` doesn't necessarily
+  ## have more elements than `s1`. That is, `s1` can be equal to `s2`.
+  runnableExamples:
+    var
+      a = initOrdSet[int]()
+      b = initOrdSet[int]()
+    a.incl(1)
+    b.incl(1); b.incl(2)
+    assert a <= b
+    a.incl(2)
+    assert a <= b
+    a.incl(3)
+    assert(not (a <= b))
+
+  for item in s1:
+    if not s2.contains(item):
+      return false
+  return true
+
+proc `<`*[A](s1, s2: OrdSet[A]): bool =
+  ## Returns true if `s1` is proper subset of `s2`.
+  ##
+  ## A strict or proper subset `s1` has all of its elements in `s2`, but `s2` has
+  ## more elements than `s1`.
+  runnableExamples:
+    var
+      a = initOrdSet[int]()
+      b = initOrdSet[int]()
+    a.incl(1)
+    b.incl(1); b.incl(2)
+    assert a < b
+    a.incl(2)
+    assert(not (a < b))
+  return s1 <= s2 and not (s2 <= s1)
+
+proc `==`*[A](s1, s2: OrdSet[A]): bool =
+  ## Returns true if both `s1` and `s2` have the same elements and set size.
+  return s1 <= s2 and s2 <= s1
+
+proc `$`*[A](s: OrdSet[A]): string =
+  ## The `$` operator for int sets.
+  ##
+  ## Converts the set `s` to a string, mostly for logging and printing purposes.
+  dollarImpl()
+
+
+
+when isMainModule:
```

Comment:
(could be done in cleanup post-merge) tests must be merged with tests/stdlib/tordsets.nim, not  in a isMainModule block; and possibly reduced in case some of it is redundant with tests/stdlib/tordsets.nim

Reply:
I carried over the main tests existing from intset but I'll migrate to tordset.nim if there is anything not covered already in those tests